src/react-native/04-intro-to-jsx/intro-to-jsx.md

```diff
@@ -201,9 +201,215 @@ Next, we want to render a list of states name in our application:
+  profession: "programmer",
+}
+
+const content = (
+  <View>
+    <Text>Hi I’m {person.name.toUpperCase()}!</Text>
+    <Text>I’m a {person.profession} living in Philadelphia.</Text>
+  </View>
+)
+```
+@highlight 8,9
+
+### JSX is a syntax extension for JavaScript
+
+Remember, JSX is an alternative syntax for normal JavaScript—it is not magic. This means that you can use JSX as a normal value, too.
+
+```tsx
+const header = <Text>Hello World</Text>
+const body = <Text>My name is {"Mike"}</Text>
+
+function MyPage() {
+  return (
+    <View>
+      {header}
+      {body}
+    </View>
+  )
+}
+```
+
+If rendered, the `screen` will output:
+
+```xml
+<View>
+  <Text>Hello World</Text>
+  <Text>My name is Mike</Text>
+</View>
+```
+
+If this surprises you, remember that underneath the syntactic sugar, JSX is nothing more than `React.createElement` calls:
+
+```tsx
+const header = React.createElement("Text", null, "Hello World")
+const body = React.createElement("Text", null, `My name is ${"Mike"}`)
+
+const page = React.createElement("View", null, [header, body])
+```
+
+### Working with conditionals and loops in JSX
+
+Only expressions that return a value may be interpolated. This includes static values, variables and calls to functions. It does not include control-flow statements such as `if`, `case`, `for`, `while`. These can either be abstracted behind a function, which is then called within the JSX or be re-written in a JSX-friendly way.
+
+To put it simply: only things that you could pass into a function can be used inside the brackets.
+
+#### Using conditionals
+
+Conditions can be rewritten using the ternary operator.
+
+```tsx
+// This does not work
+const content = (
+  <Text>
+    {
+      if (a === b) { // Control flow does not belong in JSX
+        "a and b are equal"
+      } else {
+        "a and b are different"
+      }
+    }
+  </Text>
+)
+```
+
+```tsx
+// But the same can be accomplished with ternaries
+const content = (
+  <Text>
+    {a === b // Ternaries are expressions.
+      ? "a and b are equal"
+      : "a and b are different"}
+  </Text>
+)
+```
+
+If ternaries seem excessive for any particular case, you can write all your logic in a separate function and invoke it from within JSX.
+
+```tsx
+function makeResult() {
+  return a === b ? "a and b are equal" : "a and b are different"
+}
+
+const content = <Text>{makeResult()}</Text>
+```
+
+#### Using loops
+
+JSX does not support traditional loop statements like `for`, `while`, or `do...while` directly within JSX.
+
+The example below will **not** work:
+
+```tsx
+const names = ['Alfa', 'Bravo', 'Charlie'];
+
+// This does not work
+const content = (
+  <View>
+    {
+      // Control flow does not belong in JSX
+      for (let name of names) {
+        <Text>name</Text>
+      }
+    }
+  </View>
+)
+```
+
+The `Array.map()` function is the most common and idiomatic way to render lists in JSX.
+It’s especially useful for rendering arrays of data as xml elements.
```

Comment:
`xml` -> `XML`

Or maybe this should be rephrased? Maybe “xml elements” should be “components” instead.